library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java

```diff
@@ -566,6 +568,64 @@ public void onFailure(Call call, Throwable t) {
         });
     }
 
+    /**
+     * hides the channel from queryChannels for the user until a message is added
+     *
+     * @param channel  the channel needs to hide
+     * @param callback the result callback
+     */
+    public void hideChannel(@NonNull Channel channel, @NotNull ShowHideChannelCallback callback) {
+        mService.hideChannel(channel.getType(), channel.getId(), apiKey, clientID,
+                new HideChannelRequest(user.getId()))
+                .enqueue(new Callback<ShowHideChannelResponse>() {
+                    @Override
+                    public void onResponse(Call<ShowHideChannelResponse> call, Response<ShowHideChannelResponse> response) {
+                        if (response.isSuccessful()) {
+                            callback.onSuccess(response.body());
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Call<ShowHideChannelResponse> call, Throwable t) {
+                        if (t instanceof ErrorResponse) {
+                            callback.onError(t.getMessage(), ((ErrorResponse) t).getCode());
+                        } else {
+                            callback.onError(t.getLocalizedMessage(), -1);
+                        }
+                    }
+                });
+    }
+
+    /**
+     * removes the hidden status for a channel
+     *
+     * @param channel  the channel needs to show
+     * @param callback the result callback
+     */
+    public void showChannel(@NonNull Channel channel, @NotNull ShowHideChannelCallback callback) {
+        mService.showChannel(channel.getType(), channel.getId(), apiKey, clientID,
+                new HideChannelRequest(user.getId()))
+                .enqueue(new Callback<ShowHideChannelResponse>() {
+                    @Override
+                    public void onResponse(Call<ShowHideChannelResponse> call, Response<ShowHideChannelResponse> response) {
+                        if (response.isSuccessful()) {
```

Comment:
I don't think we need to check if response is successful, that's already done. Please have a look at the `RetrofitClient` class, I might be wrong about this.